### PR TITLE
Set up Storybook interaction testing

### PR DESCRIPTION
**Pull request by manuelblum**
Run Storybook stories as [interaction tests](https://storybook.js.org/docs/writing-tests/interaction-testing) in CI via [`@storybook/addon-vitest`](https://storybook.js.org/docs/writing-tests/integrations/vitest-addon), using [Vitest browser mode](https://vitest.dev/guide/browser/) with Playwright + Chromium.

Configured in `packages/admin/admin`, `packages/admin/cms-admin`, `packages/mail-react` and `storybook/`. Each package ships a sample [`play`](https://storybook.js.org/docs/writing-stories/play-function) assertion so the wiring is exercised; follow-ups will add real coverage.

Every story is now rendered as part of the test suite — any runtime error in a story (failed render, thrown exception) surfaces as a failing CI check, so broken stories can no longer slip through.

### Sample Story with Interaction testing
473928481-2f0d5df7-549a-4456-b19a-cfcb94657c6f

--- 

CI uses the smaller [`chromium-headless-shell`](https://playwright.dev/docs/browsers#chromium-headless-shell) build to keep install time down.

> [!IMPORTANT]
> **CI test job runtime increases from ~3 min to ~13 min** due to the browser-based interaction tests.

[COM-2216]: https://vivid-planet.atlassian.net/browse/COM-2216?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

----

Jira Ticket: [COM-2216](https://vivid-planet.atlassian.net/browse/COM-2216)

### CODE REVIEW

File: docs/netlify.toml [hunk 1]
```diff
@@ -2,3 +2,6 @@
   command = "cd .. && pnpm install --frozen-lockfile && pnpm setup:ci && pnpm build:docs"
   publish = "build"
   ignore = "git diff --quiet $CACHED_COMMIT_REF $COMMIT_REF -- . ../storybook/ ../packages/admin/"
+
+[build.environment]
```
Comment (manuelblum) [inline docs/netlify.toml:6]: we should rebase when this is merged: https://github.com/vivid-planet/comet/pull/5563

File: storybook/src/admin/form/RouterTabsInFormWithSubroutes.stories.tsx [hunk 2]
```diff
@@ -19,6 +19,7 @@ function Path() {
 export default {
     title: "@comet/admin/form",
     decorators: [storyRouterDecorator()],
+    tags: ["!test"],
```
Comment (manuelblum) [inline storybook/src/admin/form/RouterTabsInFormWithSubroutes.stories.tsx]: this should be removed.

File: package.json [hunk 3]
```diff
@@ -85,7 +85,8 @@
         "peerDependencyRules": {
             "allowedVersions": {
                 "react-final-form>react": "19",
-                "react-final-form-arrays>react": "19"
+                "react-final-form-arrays>react": "19",
+                "@vitest/mocker>vite": "5 || 6 || 7"
```
Comment (manuelblum) [inline package.json]: thats required

@vitest/mocker@4 requires vite 6 or 7, but site-react/site-nextjs still use vite 5 — this override the peer range so pnpm install doesn't fail.

Comment (johnnyomair) [reply package.json]: Should be fixed by https://github.com/vivid-planet/comet/pull/5564.

File: packages/admin/admin/vitest.config.ts [hunk 4]
```diff
@@ -1,14 +1,38 @@
...
+                    name: "unit",
+                    environment: "jsdom",
+                    setupFiles: "./vitest.setup.ts",
+                },
+            },
+            {
+                plugins: [tsconfigPaths(), storybookTest({ configDir: ".storybook" })],
+                optimizeDeps: {
+                    include: ["storybook/internal/csf", "@storybook/react-vite"],
+                },
```
Comment (johnnyomair) [inline packages/admin/admin/vitest.config.ts:25]: Why is this necessary?

File: packages/admin/cms-admin/vitest.config.ts [hunk 5]
```diff
@@ -1,14 +1,56 @@
...
+                        "storybook/internal/csf",
+                        "@storybook/react-vite",
+                        "react/jsx-dev-runtime",
+                        "@mui/material",
+                        "@mui/system",
+                        "@mui/x-date-pickers",
+                        "@mui/x-date-pickers/AdapterDateFns",
+                        "date-fns/locale",
+                        "react-intl",
+                    ],
```
Comment (johnnyomair) [inline packages/admin/cms-admin/vitest.config.ts]: Why do we need to include more dependencies here than for `@comet/admin`?

**Review by johnnyomair — COMMENTED** [review]
Nice! 👏🏼 We'll have to do something about the test run time though. 😅

File: packages/admin/admin/vitest.config.ts (hunk 4 above)
Comment (manuelblum) [reply packages/admin/admin/vitest.config.ts:25]: as far as i understood it, it is necessary


**ai explains this as:**
> @storybook/addon-vitest runs stories through virtual modules that import storybook/internal/csf and @storybook/react-vite — imports Vite can't see during its static dependency crawl. Without listing them in optimizeDeps.include, Vite discovers them mid-test, re-bundles, and triggers a reload that breaks the in-flight run.

Comment (johnnyomair) [reply packages/admin/admin/vitest.config.ts:25]: We should probably reduce the list for `@comet/cms-admin` if possible (see https://github.com/vivid-planet/comet/pull/5561#discussion_r3226901445).

File: demo/api/package.json [hunk 6]
```diff
@@ -38,6 +38,7 @@
         "start:prod": "node dist/main",
         "storybook": "storybook dev -p 4004 --no-open --disable-telemetry",
         "test": "pnpm api-generator && vitest --run",
+        "test:unit": "pnpm api-generator && vitest --run",
```
Comment (johnnyomair) [inline demo/api/package.json:41]: Do we need dedicated test:unit scripts which duplicate test? IMO a common practice is test for unit tests and test:e2e for end-to-end tests.

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -43,13 +43,60 @@ jobs:
...
+                  registry-url: "https://registry.npmjs.org"
+                  cache: "pnpm"
+
+            - run: pnpm install --frozen-lockfile
+
+            - name: Setup project for CI run
+              run: pnpm run setup:ci
+
+            - name: Build
+              run: pnpm run build:packages
```
Comment (johnnyomair) [inline .github/workflows/test.yml]: I believe it should be possible to shared these steps for both "jobs". 🤔

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -43,13 +43,60 @@ jobs:
...
+            - name: Test (e2e)
+              run: pnpm run test:e2e
+
+            - name: Upload test results
+              uses: actions/upload-artifact@v7
+              if: success() || failure()
+              with:
+                  name: test-results-e2e
+                  path: |
+                      packages/**/junit.xml
```
Comment (johnnyomair) [inline .github/workflows/test.yml]: Won't this upload unit test results as well?

File: packages/admin/cms-admin/package.json [hunk 9]
```diff
@@ -42,6 +42,8 @@
         "start:types": "tsc --project ./tsconfig.json --emitDeclarationOnly --watch --preserveWatchOutput",
         "storybook": "storybook dev -p 26647 --no-open --disable-telemetry",
         "test": "vitest --run",
+        "test:e2e": "vitest --run --project storybook",
```
Comment (nsams) [inline packages/admin/cms-admin/package.json]: I would not call storybook tests "e2e".

To me an e2e test does test a complete application with no mocks. Nothing a cms-admin library can do. (there could (should) be e2e tests in Demo tough)

we could call it test:storybook?

Comment (manuelblum) [reply packages/admin/cms-admin/package.json]: Good point — renamed to `test:storybook` across all packages and the CI job in 7ab8858a8.

File: packages/admin/cms-admin/vitest.config.ts (hunk 5 above)
Comment (manuelblum) [reply packages/admin/cms-admin/vitest.config.ts]: Trimmed the list in https://github.com/vivid-planet/comet/pull/5561/commits/371991cee — locally it works with the new commit. Pipelines are right now broken on an unrelated codeload.github.com hiccup, will re-run once it clears.

Comment (manuelblum) [reply packages/admin/cms-admin/vitest.config.ts]: pipeline worked and succeeded again, and it looks good so far

File: packages/admin/admin/vitest.config.ts (hunk 4 above)
Comment (manuelblum) [reply packages/admin/admin/vitest.config.ts:25]: handled here: https://github.com/vivid-planet/comet/pull/5561#discussion_r3303720078

File: .github/workflows/test.yml (hunk 7 above)
Comment (manuelblum) [reply .github/workflows/test.yml]: some of those steps got repeated in multiple workflows:

[.github/workflows/build-docs.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/build-docs.yml)
[.github/workflows/chromatic.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/chromatic.yml)
[.github/workflows/copilot-setup-steps.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/copilot-setup-steps.yml)
[.github/workflows/i18n.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/i18n.yml)
[.github/workflows/lint.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/lint.yml)
[.github/workflows/release.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/release.yml)
[.github/workflows/release-beta.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/release-beta.yml)
[.github/workflows/release-canary.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/release-canary.yml)
[.github/workflows/test.yml](https://github.com/vivid-planet/comet/blob/thirsty-lamarr-7be25e/.github/workflows/test.yml)


i will try to make parts reusable in another pull requeast.

Comment (manuelblum) [reply .github/workflows/test.yml]: did this in this pull request: https://github.com/vivid-planet/comet/pull/5715

I would recommand to merge the pull request #5715 first.

File: .github/workflows/test.yml (hunk 8 above)
Comment (manuelblum) [reply .github/workflows/test.yml]: In practice no — the jobs run since some commits on isolated runners so nothing else writes `junit.xml` there. But the glob didn't enforce it. Split the junit output per project (https://github.com/vivid-planet/comet/pull/5561/commits/be961d5f5) so `test:unit` writes `junit-unit.xml`, `test:storybook` writes `junit-storybook.xml`, and the upload globs target each name explicitly.

File: .github/workflows/test.yml (hunk 7 above)
Comment (manuelblum) [reply .github/workflows/test.yml]: Done in https://github.com/vivid-planet/comet/pull/5561/commits/44e1d8af4 — both jobs now go through `./.github/actions/setup` (introduced in #5715, which this branch is rebased onto).

File: demo/api/package.json (hunk 6 above)
Comment (manuelblum) [reply demo/api/package.json:41]: I added the `test:unit` script in all packages. `test:unit` and `test:storybook` are used in CI/CD, as we aligned to split unit and storybook test runs into separate jobs.

The `test` script, in my opinion, should be responsible for grouping its sub-scripts. Thats the pattern I thought we want to follow.

Like we do in the root package:

`dev` → `pnpm dev:admin && pnpm dev:cms && …`
`lint` → `run-p lint:eslint lint:prettier lint:tsc`

---

Just stumbled across some patterns in the scripts where we are not 100% following this pattern (e.g. `lint:fix`, `lint:ci` )

File: .github/workflows/test.yml [hunk 10]
```diff
@@ -30,15 +30,49 @@ jobs:
...
               uses: actions/upload-artifact@v7
               if: success() || failure()
               with:
-                  name: test-results
+                  name: test-results-unit
                   path: |
-                      packages/*/junit.xml
-                      packages/*/*/junit.xml
+                      packages/**/junit-unit.xml
+                      packages/**/junit.xml
```
Comment (johnnyomair) [inline .github/workflows/test.yml]: This would once again report test results in node_modules/. I believe you meant this:

```suggestion
                      packages/*/junit-unit.xml
                      packages/*/*/junit-unit.xml
```

File: .github/workflows/test.yml [hunk 11]
```diff
@@ -30,15 +30,49 @@ jobs:
...
+            - name: Test (storybook)
+              run: pnpm run test:storybook
+
+            - name: Upload test results
+              uses: actions/upload-artifact@v7
+              if: success() || failure()
+              with:
+                  name: test-results-storybook
+                  path: |
+                      packages/**/junit-storybook.xml
```
Comment (johnnyomair) [inline .github/workflows/test.yml]: Change this to not report test results in node_modules:

```suggestion
                      packages/*/junit-storybook.xml
                      packages/*/*/junit-storybook.xml
```

File: .github/workflows/test.yml (hunk 8 above)
Comment (johnnyomair) [reply .github/workflows/test.yml]: If we change this we'll have to update the pattern in test-report as well: https://github.com/vivid-planet/comet/blob/com-2216/storybook-interaction-tests/.github/workflows/test-report.yml#L15.

I'm also fine with reverting the change as it isn't a problem in practice.

File: demo/api/package.json (hunk 6 above)
Comment (johnnyomair) [reply demo/api/package.json:41]: What about `test:watch` scripts? Will they run unit tests or Storybook tests?

Comment (manuelblum) [reply demo/api/package.json:41]: maybe we should reflect our rules (if we have some 😅) and cleanup all package scripts in a separat pull request?

File: .github/workflows/test.yml (hunk 8 above)
Comment (manuelblum) [reply .github/workflows/test.yml]: i think we already discussed and changed this

Comment (johnnyomair) [reply .github/workflows/test.yml]: No, not completely. The test workflow now emits junit-unit.xml and junit-storybook.xml files, but the test-report (a separate workflow) searches for junit.xml files. This will result in no files being reported. So we have to update the pattern here: https://github.com/vivid-planet/comet/blob/com-2216/storybook-interaction-tests/.github/workflows/test-report.yml#L15.

File: packages/admin/cms-admin/vitest.config.ts (hunk 5 above)
Comment (johnnyomair) [reply packages/admin/cms-admin/vitest.config.ts]: Commit 7621aee6503fad9187c5ca6db48d6dee29678152 now re-adds the list. I'm not sure why we need it, might point to an incorrect dependency setup. It's fine if it's only required for the Storybook tests.

**Review by johnnyomair — DISMISSED** [review]

File: packages/admin/cms-admin/vitest.config.ts (hunk 5 above)
Comment (manuelblum) [reply packages/admin/cms-admin/vitest.config.ts]: i cleaned up git history - it was already too confusing to me - it should not be there anymore.

**Review by johnnyomair — DISMISSED** [review]

**Comment by manuelblum** [comment]
Rebased onto latest `main` to resolve merge conflicts.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by nsams — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]
